### ISSUE TITLE
Error on cake with gazeDeviceManager

### ISSUE THREAD

**Issue by kuhlkrein** [schwabdidier/GazePlay#452]
I think this error is related to error #427 .
For what I see, it seems that sometimes, when we try to remove a eventfilter from the device manager, we are returning a null pointer because the function onGazeUpdate try to access something that is becoming null during it's execution (I'm not sure about that)

**Comment by kuhlkrein** [comment]
`2018-06-05 09:34:22,149 ERROR - java.lang.ArrayIndexOutOfBoundsException: -1
2018-06-05 09:34:22,149 ERROR - 	at java.util.ArrayList.elementData(ArrayList.java:418)
2018-06-05 09:34:22,149 ERROR - 	at java.util.ArrayList.get(ArrayList.java:431)
2018-06-05 09:34:22,149 ERROR - 	at com.sun.javafx.collections.ObservableListWrapper.get(ObservableListWrapper.java:89)
2018-06-05 09:34:22,149 ERROR - 	at com.sun.javafx.collections.VetoableListDecorator.get(VetoableListDecorator.java:306)
2018-06-05 09:34:22,149 ERROR - 	at javafx.scene.Parent.updateCachedBounds(Parent.java:1591)
2018-06-05 09:34:22,149 ERROR - 	at javafx.scene.Parent.recomputeBounds(Parent.java:1535)
2018-06-05 09:34:22,149 ERROR - 	at javafx.scene.Parent.impl_computeGeomBounds(Parent.java:1388)
2018-06-05 09:34:22,149 ERROR - 	at javafx.scene.layout.Region.impl_computeGeomBounds(Region.java:3078)
2018-06-05 09:34:22,149 ERROR - 	at javafx.scene.Node.updateGeomBounds(Node.java:3579)
2018-06-05 09:34:22,150 ERROR - 	at javafx.scene.Node.getGeomBounds(Node.java:3532)
2018-06-05 09:34:22,150 ERROR - 	at javafx.scene.Node.getLocalBounds(Node.java:3480)
2018-06-05 09:34:22,150 ERROR - 	at javafx.scene.Node.updateTxBounds(Node.java:3643)
2018-06-05 09:34:22,150 ERROR - 	at javafx.scene.Node.getTransformedBounds(Node.java:3426)
2018-06-05 09:34:22,150 ERROR - 	at javafx.scene.Node.updateBounds(Node.java:559)
2018-06-05 09:34:22,150 ERROR - 	at javafx.scene.Parent.updateBounds(Parent.java:1719)
2018-06-05 09:34:22,150 ERROR - 	at javafx.scene.Parent.updateBounds(Parent.java:1717)
2018-06-05 09:34:22,150 ERROR - 	at javafx.scene.Parent.updateBounds(Parent.java:1717)
2018-06-05 09:34:22,150 ERROR - 	at javafx.scene.Parent.updateBounds(Parent.java:1717)
2018-06-05 09:34:22,150 ERROR - 	at javafx.scene.Parent.updateBounds(Parent.java:1717)
2018-06-05 09:34:22,150 ERROR - 	at javafx.scene.Parent.updateBounds(Parent.java:1717)
2018-06-05 09:34:22,150 ERROR - 	at javafx.scene.Parent.updateBounds(Parent.java:1717)
2018-06-05 09:34:22,150 ERROR - 	at javafx.scene.Scene$ScenePulseListener.pulse(Scene.java:2404)
2018-06-05 09:34:22,150 ERROR - 	at com.sun.javafx.tk.Toolkit.lambda$runPulse$30(Toolkit.java:355)
2018-06-05 09:34:22,150 ERROR - 	at java.security.AccessController.doPrivileged(Native Method)
2018-06-05 09:34:22,150 ERROR - 	at com.sun.javafx.tk.Toolkit.runPulse(Toolkit.java:354)
2018-06-05 09:34:22,151 ERROR - 	at com.sun.javafx.tk.Toolkit.firePulse(Toolkit.java:381)
2018-06-05 09:34:22,151 ERROR - 	at com.sun.javafx.tk.quantum.QuantumToolkit.pulse(QuantumToolkit.java:510)
2018-06-05 09:34:22,151 ERROR - 	at com.sun.javafx.tk.quantum.QuantumToolkit.pulse(QuantumToolkit.java:490)
2018-06-05 09:34:22,151 ERROR - 	at com.sun.javafx.tk.quantum.QuantumToolkit.lambda$runToolkit$404(QuantumToolkit.java:319)
2018-06-05 09:34:22,151 ERROR - 	at com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:95)
2018-06-05 09:34:22,151 ERROR - 	at com.sun.glass.ui.win.WinApplication._runLoop(Native Method)
2018-06-05 09:34:22,151 ERROR - 	at com.sun.glass.ui.win.WinApplication.lambda$null$148(WinApplication.java:191)
2018-06-05 09:34:22,151 ERROR - 	at java.lang.Thread.run(Thread.java:745)
2018-06-05 09:34:22,160 ERROR - java.util.concurrent.FutureTask.run(FutureTask.java:266)
2018-06-05 09:34:22,161 ERROR - Exception in thread "JavaFX Application Thread" java.lang.ArrayIndexOutOfBoundsException
2018-06-05 09:34:22,164 ERROR - Exception in thread "JavaFX Application Thread" java.lang.ArrayIndexOutOfBoundsException
2018-06-05 09:34:22,165 ERROR - javafx.concurrent.Service.lambda$null$492(Service.java:725)
2018-06-05 09:34:22,165 ERROR - java.security.AccessController.doPrivileged(Native Method)
2018-06-05 09:34:22,165 ERROR - javafx.concurrent.Service.lambda$executeTask$493(Service.java:724)
2018-06-05 09:34:22,165 ERROR - java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
2018-06-05 09:34:22,166 ERROR - java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
2018-06-05 09:34:22,167 ERROR - java.lang.Thread.run(Thread.java:745)
2018-06-05 09:34:22,168 ERROR - Exception on Gaze position update : {}
java.lang.ArrayIndexOutOfBoundsException: null
2018-06-05 09:34:22,168 ERROR - Exception on Gaze position update : {}
java.lang.ArrayIndexOutOfBoundsException: null
2018-06-05 09:34:22,171 ERROR - Exception on Gaze position update : {}
java.lang.ArrayIndexOutOfBoundsException: null
2018-06-05 09:34:22,171 ERROR - Exception on Gaze position update : {}
java.lang.ArrayIndexOutOfBoundsException: null
2018-06-05 09:34:22,172 ERROR - Exception on Gaze position update : {}
java.lang.ArrayIndexOutOfBoundsException: null
2018-06-05 09:34:22,173 ERROR - Exception on Gaze position update : {}
java.lang.ArrayIndexOutOfBoundsException: null
2018-06-05 09:34:22,173 ERROR - Exception on Gaze position update : {}
java.lang.ArrayIndexOutOfBoundsException: null
2018-06-05 09:34:22,174 ERROR - Exception on Gaze position update : {}
java.lang.ArrayIndexOutOfBoundsException: null
2018-06-05 09:34:22,174 ERROR - Exception on Gaze position update : {}
java.lang.ArrayIndexOutOfBoundsException: null
2018-06-05 09:34:22,174 ERROR - Exception on Gaze position update : {}
java.lang.ArrayIndexOutOfBoundsException: null
2018-06-05 09:34:22,175 ERROR - Exception on Gaze position update : {}
java.lang.ArrayIndexOutOfBoundsException: null
2018-06-05 09:34:22,175 ERROR - Exception on Gaze position update : {}
java.lang.ArrayIndexOutOfBoundsException: null
2018-06-05 09:34:22,177 ERROR - Exception on Gaze position update : {}
java.lang.ArrayIndexOutOfBoundsException: null
2018-06-05 09:34:22,177 ERROR - Exception on Gaze position update : {}`


@schwabdidier , @YannRobert , any idea ? I'm locked on this problem since many days on this game and gaze menu... it seems that it's due to gazeDeviceManager and gazeEvent but I don't find any answer

**Comment by kuhlkrein** [comment]
```
java.lang.NullPointerException: null
	at com.sun.javafx.text.PrismTextLayout.addTextRun(PrismTextLayout.java:755)
	at com.sun.javafx.text.GlyphLayout.addTextRun(GlyphLayout.java:140)
	at com.sun.javafx.text.GlyphLayout.breakRuns(GlyphLayout.java:312)
	at com.sun.javafx.text.PrismTextLayout.buildRuns(PrismTextLayout.java:770)
	at com.sun.javafx.text.PrismTextLayout.layout(PrismTextLayout.java:1021)
	at com.sun.javafx.text.PrismTextLayout.ensureLayout(PrismTextLayout.java:223)
	at com.sun.javafx.text.PrismTextLayout.getBounds(PrismTextLayout.java:246)
	at javafx.scene.text.Text.getLogicalBounds(Text.java:358)
	at javafx.scene.text.Text.impl_computeGeomBounds(Text.java:1168)
	at javafx.scene.Node.updateGeomBounds(Node.java:3579)
	at javafx.scene.Node.getGeomBounds(Node.java:3532)
	at javafx.scene.Node.getLocalBounds(Node.java:3480)
	at javafx.scene.Node.updateTxBounds(Node.java:3643)
	at javafx.scene.Node.getTransformedBounds(Node.java:3426)
	at javafx.scene.Parent.createCachedBounds(Parent.java:1691)
	at javafx.scene.Parent.updateCachedBounds(Parent.java:1554)
	at javafx.scene.Parent.recomputeBounds(Parent.java:1535)
	at javafx.scene.Parent.impl_computeGeomBounds(Parent.java:1388)
	at javafx.scene.layout.Region.impl_computeGeomBounds(Region.java:3078)
	at javafx.scene.Node.updateGeomBounds(Node.java:3579)
	at javafx.scene.Node.getGeomBounds(Node.java:3532)
	at javafx.scene.Node.getLocalBounds(Node.java:3480)
	at javafx.scene.Node.updateTxBounds(Node.java:3643)
	at javafx.scene.Node.getTransformedBounds(Node.java:3426)
	at javafx.scene.Parent.getChildTransformedBounds(Parent.java:1732)
	at javafx.scene.Parent.recomputeBounds(Parent.java:1524)
	at javafx.scene.Parent.impl_computeGeomBounds(Parent.java:1388)
	at javafx.scene.layout.Region.impl_computeGeomBounds(Region.java:3078)
	at javafx.scene.Node.updateGeomBounds(Node.java:3579)
	at javafx.scene.Node.getGeomBounds(Node.java:3532)
	at javafx.scene.Node.getLocalBounds(Node.java:3480)
	at javafx.scene.Node.updateTxBounds(Node.java:3643)
	at javafx.scene.Node.getTransformedBounds(Node.java:3426)
	at javafx.scene.Parent.getChildTransformedBounds(Parent.java:1732)
	at javafx.scene.Parent.updateCachedBounds(Parent.java:1596)
	at javafx.scene.Parent.recomputeBounds(Parent.java:1535)
	at javafx.scene.Parent.impl_computeGeomBounds(Parent.java:1388)
	at javafx.scene.layout.Region.impl_computeGeomBounds(Region.java:3078)
	at javafx.scene.Node.updateGeomBounds(Node.java:3579)
	at javafx.scene.Node.getGeomBounds(Node.java:3532)
	at javafx.scene.Node.getLocalBounds(Node.java:3480)
	at javafx.scene.Node.containsBounds(Node.java:3886)
	at javafx.scene.Node.contains(Node.java:3862)
	at javafx.scene.Node.contains(Node.java:3916)
	at net.gazeplay.commons.gaze.devicemanager.AbstractGazeDeviceManager.onGazeUpdate(AbstractGazeDeviceManager.java:116)
	at net.gazeplay.commons.gaze.devicemanager.TobiiGazeDeviceManager$1$1.call(TobiiGazeDeviceManager.java:47)
	at net.gazeplay.commons.gaze.devicemanager.TobiiGazeDeviceManager$1$1.call(TobiiGazeDeviceManager.java:1)
	at javafx.concurrent.Task$TaskCallable.call(Task.java:1423)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at javafx.concurrent.Service.lambda$null$492(Service.java:725)
	at java.security.AccessController.doPrivileged(Native Method)
	at javafx.concurrent.Service.lambda$executeTask$493(Service.java:724)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by kuhlkrein** [comment]
fixed by #486
